test/endpoint_testrunner.py

```diff
@@ -0,0 +1,238 @@
+import re
+import git
+import sys
+import logging
+from multiprocessing import Pool
+from packaging.version import Version #, parse
+from junit_xml import TestSuite, TestCase
+from invoke.context import Context
+
+PUBLIC_DLC_REGISTRY = "763104351884"
+PUBLIC_DLC_REGION = "us-west-2"
+REPORT_XML = os.path.join(os.getcwd(),"test", "sagemaker_endpoint_tests.xml")
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+LOGGER.addHandler(logging.StreamHandler(sys.stderr))
+
+def generate_report(report, test_name, test_class, test_out, test_err=''):
+    """
+    Generate junitxml test report
+    :param report: CodeBuild Report
+    Returns: None
+    """
+    # TestCase(name, classname, test time, system-out, system-err)
+    test_cases = [TestCase(test_name, test_class, 1, test_out, test_err)]
+    ts = TestSuite(report, test_cases)
+    with open(report, "w") as report_file:
+        TestSuite.to_file(report_file, [ts], prettyprint=False)
+
+
+def get_canary_default_tag_py3_version(framework, version):
+    """
+    Currently, only TF2.2 images and above have major/minor python version in their canary tag. Creating this function
+    to conditionally choose a python version based on framework version ranges. If we move up to py38, for example,
+    this is the place to make the conditional change.
+    :param framework: tensorflow1, tensorflow2, mxnet, pytorch
+    :param version: fw major.minor version, i.e. 2.2
+    :return: default tag python version
+    """
+    if framework == "tensorflow":
+        if Version("2.2") <= Version(version) < Version("2.6"):
+            return "py37"
+        if Version("2.6") <= Version(version) < Version("2.8"):
+            return "py38"
+        if Version(version) >= Version("2.8"):
+            return "py39"
+
+    if framework == "mxnet":
+        if Version(version) == Version("1.8"):
+            return "py37"
+        if Version(version) >= Version("1.9"):
+            return "py38"
+
+    if framework == "pytorch":
+        if Version("1.9") <= Version(version) < Version("1.13"):
+            return "py38"
+        if Version(version) >= Version("1.13") and Version(version) < Version("2.0"):
+            return "py39"
+        if Version(version) >= Version("2.0"):
+            return "py310"
+
+    return "py3"
+
+def get_sagemaker_images_from_github(registry, framework, region, image_type, processor=None):
+    """
+    Return which Inference GPU canary images to run canary tests on for a given framework and AWS region
+
+    :param framework: ML framework (mxnet, tensorflow, pytorch)
+    :param region: AWS region
+    :param image_type: inference
+    :return: a list of space separated image uris
+    """
+    customer_type_tag = "-sagemaker"
+    version_regex = {
+        "tensorflow": rf"tf(-sagemaker)?{customer_type_tag}-(\d+.\d+)",
+        "mxnet": rf"mx(-sagemaker)?{customer_type_tag}-(\d+.\d+)",
+        "pytorch": rf"pt(-sagemaker)?{customer_type_tag}-(\d+.\d+)",
```

Comment:
Please add a TODO to include all SM-DLCs, and not just TF, PT, MX. Or is there a reason to only test these 3 frameworks in SM canaries?